src/main/java/organice/logic/commands/DoneCommand.java

```diff
@@ -71,26 +78,58 @@ public boolean isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model mo
         }
         if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
                 && match(donor, patient)) {
-            //todo check if tasklist isempty
-            return true;
+            return donor.getStatus().isProcessing() && patient.getStatus().isProcessing();
         } else {
             return false;
         }
+    }
 
+    /**
+     * A boolean to determine the value of the String, result
+     * @param result a String from the user input
+     * @return true if the result shows a pass, false if the result shows a fail.
+     */
+    public boolean isPass(String result) {
+        if (result.toLowerCase().equals("pass")) {
+            return true;
+        } else if (result.toLowerCase().equals("fail")) {
```

Comment:
I think you could add an assert statement or throw an error at the 'else' condition, because the only valid inputs are 'pass' and 'fail'.

Reply:
done thank you :))